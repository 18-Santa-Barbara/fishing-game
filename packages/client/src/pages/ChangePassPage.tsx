import { TextField, Button, Avatar, Box, Typography } from '@mui/material';
import { Component, ReactNode } from 'react';
import { Link } from 'react-router-dom';
import { withStyles } from '@mui/styles';
import { AccountCircle } from '@mui/icons-material';
import { apiRequestPut } from '../utils/api';
import { API } from '../utils/constants';
import withNavigation from '../hocs/with-navigation/WithNavigation';

interface SignUpState {
  error: string;
  oldPassword: string;
  newPassword: string;
  confirm: string;
}

const styles = {
  paper: {
    textAlign: 'center',
    boxShadow: '0px 0px 6px rgb(0 0 0 / 14%)',
    borderRadius: '12px',
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    padding: '16px',
  },
  btn: {
    margin: '12px 0',
    width: '33%',
  },
};

class SignUp extends Component {
  state: SignUpState = {
    oldPassword: '',
    newPassword: '',
    confirm: '',
    error: '',
  };

  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const {
      target: { name, value },
    } = e;
    this.setState({ [name]: value });
  };

  submit = () => {
    const { oldPassword, newPassword } = this.state;
    apiRequestPut(`${API}/user/password`, { oldPassword, newPassword })
      .then(res => {
        if ('reason' in res) {
          this.setState({ error: res.reason });
        } else {
          this.props.navigate('/profile');
        }
      })
      .catch(() => this.setState({ error: 'Ошибка!' }));
  };

  render(): ReactNode {
    const { oldPassword, newPassword, confirm, error } = this.state;
    const { classes } = this.props;
    return (
      <Box className={classes.paper} maxWidth={'md'}>
        <div style={{ display: 'flex', justifyContent: 'center' }}>
          <Avatar
            sx={{ m: 1, bgcolor: 'primary', width: '60px', height: '60px' }}>
            <AccountCircle />
          </Avatar>
        </div>
        <div>
          <TextField
            name="oldPassword"
            variant="outlined"
            label="Old password"
            margin="normal"
            value={oldPassword}
            autoFocus
            onChange={this.handleChange}
            fullWidth
          />
          <TextField
            name="newPassword"
            onChange={this.handleChange}
            variant="outlined"
            value={newPassword}
            label="New password"
            margin="normal"
            fullWidth
          />
          <TextField
            name="confirm"
            variant="outlined"
            label="Confirm password"
            margin="normal"
            value={confirm}
            onChange={this.handleChange}
            fullWidth
          />
          {error && (
            <Typography className={classes.err} variant="h6">
              {error || 'Ошибка!'}
            </Typography>
          )}
          <Button
            type="submit"
            fullWidth
            onClick={this.submit}
            variant="contained"
            color="primary"
            className={classes.btn}>
            Sign Up
          </Button>
        </div>
        <Link to={'/profile'}>Back to profile</Link>
      </Box>
    );
  }
}

export default withStyles(styles)(withNavigation(SignUp));
