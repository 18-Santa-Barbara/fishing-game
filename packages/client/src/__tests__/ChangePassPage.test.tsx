import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import * as renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';

import ChangePassPage from '../pages/ChangePassPage';

it('ChangePassPage page', () => {
  const mockStore = configureStore();
  const store = mockStore({ userApi: 'test' });
  const tree = renderer
    .create(
      <MemoryRouter>
        <Provider store={store}>
          <ChangePassPage />
        </Provider>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
