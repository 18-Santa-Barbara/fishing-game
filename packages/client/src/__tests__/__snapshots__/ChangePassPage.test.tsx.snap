// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassPage page 1`] = `
<div
  className="Component-paper-1 MuiBox-root css-10rvlso"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  />
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Old password
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={true}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          name="oldPassword"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Old password
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r1:"
        id=":r1:-label"
      >
        New password
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r1:"
          name="newPassword"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              New password
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r2:"
        id=":r2:-label"
      >
        Confirm password
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r2:"
          name="confirm"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Confirm password
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth Component-btn-2 css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Sign Up
    </button>
  </form>
  <a
    href="/profile"
    onClick={[Function]}
  >
    Back to profile
  </a>
</div>
`;
